src/test-runners.js

```diff
@@ -0,0 +1,161 @@
+import path from 'path'
+import { createRequire } from 'module'
+import { fileURLToPath } from 'url'
+
+const require = createRequire(import.meta.url)
+const __dirname = path.dirname(fileURLToPath(import.meta.url))
+
+/** @type {import('./types.js').TestRunner} */
+export const mocha = {
+  options: {
+    allowUncaught: false,
+    bail: true,
+    reporter: 'spec',
+    timeout: 5000,
+    color: true,
+    ui: 'bdd',
+  },
+  compileRuntime(options, paths) {
+    return `
+import mocha from 'mocha/mocha.js'
+
+const { allowUncaught, bail, reporter, timeout, color, ui, grep } =
+  process.env.PW_TEST.testRunner.options
```

Comment:
Like a path to a esm module?

Reply:
> I find it really confusing that `env` would contain anything but string values, maybe this should be something like

i dont understand this, process.env is json parsed already its a js object

Reply:
> Like a path to a esm module?

No this is what I mean

```js
/** @type {import('./types.js').TestRunner} */
export const mocha = {
  options: {
    allowUncaught: false,
    bail: true,
    reporter: 'spec',
    timeout: 5000,
    color: true,
    ui: 'bdd',
  },
  compileRuntime: (options, paths) => `
import mocha from 'mocha/mocha.js'

// no need to mess around with process.env your options are just passed in
// if tests fail promise will fail otherwise succeed.
export default (options) => new Promise((succeed, fail) => {
  mocha.setup(options)

  ${paths.map((url) => `await import('${url}')`).join('\n')}

  mocha.run(failure => failure ? fail(failure) : succeed())
`,
  },
}
```